src/__internal__/utils/helpers/tags/tags.ts

```diff
@@ -1,12 +1,21 @@
+interface Props {
+  "data-element"?: string;
+  "data-role"?: string;
+}
+
+interface TagProps extends Props {
+  "data-component": string;
+}
+
 /**
  * Builds props object containing top level data tags
  *
  * @param {String} name of component
  * @param {Object} component props
- * @return {Object} dataTagProps
+ * @return {Object} dataProps
  */
-function tagComponent(component, props) {
-  const tagProps = {
+function tagComponent(component: string, props: Props): TagProps {
+  const tagProps: TagProps = {
```

Comment:
```suggestion
  const tagProps: TagProps = {
    "data-component": component,
    ...dataProps,
  }
```

Reply:
If I'm understanding the suggestion correctly, I don't think we should make this change as it will spread the rest on to any component that uses it rather than filtering and spreading just the tag props

```
function tagComponent(componentName: string, props: DataProps): TagProps {
  return {
    "data-component": componentName,
    ...props
  };
 ```

Reply:
if indeed all properties are passed in this object, then I agree that this code does not need to be changed.
At the time of writing the comment above, the object interface looked like this:
```
interface Props {
  "data-element"?: string;
  "data-role"?: string;
}
```
so then it would made perfect sense to do that change.